src/v/storage/types.h

```diff
@@ -93,7 +103,17 @@ class snapshotable_stm {
  */
 class stm_manager {
 public:
-    void add_stm(ss::shared_ptr<snapshotable_stm> stm) { _stms.push_back(stm); }
+    template<
+      typename T,
+      typename = std::enable_if_t<std::is_base_of_v<snapshotable_stm, T>>>
```

Comment:
I don't think you need SFINAE here

Reply:
Yep. Originally thought they were going to be distinct types but they ended up having a common base. Consider this deleted practice code for me.